lib/cloudkeeper/one/opennebula/image_handler.rb

```diff
@@ -82,7 +82,12 @@ def register(image_template, datastore, group)
 
           handle_opennebula_error { image.allocate(image_template, datastore.id) }
 
-          timeout { sleep(Cloudkeeper::One::Opennebula::Handler::API_POLLING_WAIT) until ready? image }
+          timeout do
+            until ready? image
+              raise Cloudkeeper::One::Errors::Opennebula::ResourceStateError, image['TEMPLATE/ERROR'] if error? image
+              sleep(Cloudkeeper::One::Opennebula::Handler::API_POLLING_WAIT)
```

Comment:
It just looks weird ... when you compare it with the previous line. But I don't have a problem with keeping it as is.